packages/trader/src/RangeTrader.ts

```diff
@@ -0,0 +1,177 @@
+import winston from "winston";
+import Web3 from "web3";
+import BigNumber from "bignumber.js";
+const ExchangeAdapterInterface = require("./exchange-adapters/ExchangeAdapterInterface");
+
+const { ConvertDecimals, createFormatFunction, createObjectFromDefaultProps } = require("@uma/common");
+import assert from "assert";
+
+export class RangeTrader {
+  readonly normalizePriceFeedDecimals: any;
+  readonly formatDecimalString: any;
+
+  readonly BN: any;
+  readonly toBN: any;
+  readonly toWei: any;
+  readonly fromWei: any;
+
+  readonly tradeExecutionThreshold: any;
+  readonly targetPriceSpread: any;
+  readonly fixedPointAdjustment: any;
+
+  constructor(
+    /**
+   * @notice Constructs new Range Trader.
+   * @param {Object} logger Module used to send logs.
+   * @param {Object} web3 Provider from Truffle/node  to connect to Ethereum network.
+   * @param {Object} tokenPriceFeed Price feed to fetch the current synthetic token trading price. EG a Dex price feed.
+   * @param {Object} referencePriceFeed Price feed to fetch the "real" identifier price. EG a Cryptowatch price feed.
+   * @param {Object} exchangeAdapter Interface to interact with on-chain exchange. EG: Uniswap.
+   * @param {Object} rangeTraderConfig: Config to parameterize the range trader. Expected:
+   *      { tradeExecutionThreshold: 0.2,  -> error amount which must be exceeded for a correcting trade to be executed.
+            targetPriceSpread: 0.05 }      -> target price that should be present after a correcting trade has concluded.
+   */
+    readonly logger: winston.Logger,
+    readonly web3: Web3,
+    readonly tokenPriceFeed: any,
+    readonly referencePriceFeed: any,
+    readonly exchangeAdapter: typeof ExchangeAdapterInterface,
+    readonly rangeTraderConfig: any
```

Comment:
> I tried this out and made _some_ progress. first off, I was able to define the type of the `rangeTraderConfig`. 

Secondly, I did not implement your `type DefaultConfig<T> =` as I'm not actually sure what this will enforce. The object params cant be anything other than a `number` as the type in the param enforces this. Note that this was actually the case previous as the `defaultConfig` enforced hat the `isValid` method was a `number` for both params. What will your type definition above also prevent?

Reply:
Yeah, this may not be the right solution. I think defining an interface makes sense to start and I will think more about generic common functions down the road.